homeassistant/components/binary_sensor/xboxlive.py

```diff
@@ -0,0 +1,172 @@
+"""
+Sensor for Xbox Live account status.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.xbox_live/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, STATE_UNKNOWN
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+
+REQUIREMENTS = ['xbox==0.1.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_XUIDS = 'xuids'
+CONF_GAMERTAGS = 'gamertags'
+
+ICON = 'mdi:xbox'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_XUIDS): vol.All(cv.ensure_list, [cv.positive_int]),
+    vol.Required(CONF_GAMERTAGS): vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Xbox platform."""
+    import xbox
+    xbox.client.authenticate(login=config.get(CONF_USERNAME),
+                             password=config.get(CONF_PASSWORD))
+    devices = []
+
+    for xuid in config.get(CONF_XUIDS):
+        try:
+            gamer_profile = xbox.GamerProfile.from_xuid(xuid)
+        except xbox.exceptions.GamertagNotFound:
+            _LOGGER.error("Can not found Xbox UID %s", xuid)
+            continue
+        devices.append(XboxSensor(hass, gamer_profile))
+
+    for gamertag in config.get(CONF_GAMERTAGS):
+        try:
+            gamer_profile = xbox.GamerProfile.from_gamertag(gamertag)
+        except xbox.exceptions.GamertagNotFound:
+            _LOGGER.error("Can not found Gamertag %s", gamertag)
+            continue
+        devices.append(XboxSensor(hass, gamer_profile))
+
+    if devices:
+        add_devices(devices)
+    else:
+        return False
+
+
+class XboxSensor(BinarySensorDevice):
+    """A class for the Xbox account."""
+
+    def __init__(self, hass, gamer_profile):
```

Comment:
Don't pass in `hass`, it will be set on the entity when it has been added to home assistant.